torchtext/csrc/basic_english_normalize.cpp

```diff
@@ -0,0 +1,49 @@
+#include "basic_english_normalize.h"
+#include <sstream>
+
+namespace torchtext {
+
+BasicEnglishNormalize::BasicEnglishNormalize() {
+  for (const auto &pattern : patterns_) {
+    regex_objects_.push_back(Regex(pattern));
```

Comment:
Since we have static patterns, I wonder if there are static initializers for RE2 regex objects that pre-compile the regex at compile time.

Reply:
For example, see here: https://chromium.googlesource.com/external/github.com/google/re2/+/refs/heads/master/re2/re2.h#132

```

// Example: precompile pattern for faster matching:
--
  | //    RE2 pattern("h.*o");
  | //    while (ReadLine(&str)) {
  | //      if (RE2::FullMatch(str, pattern)) ...;
  | //    }
  | //


```

Reply:
Updated this to construct precompiled patterns during construction. Lmk what you think!